### ISSUE TITLE
Expose isTraceEnabled etc.

### ISSUE THREAD

**Issue by ngocdaothanh** [mattroberts297/slf4s#7]
I'm using slf4s 1.1.7.

I want to do like this:
- If trace is enabled, log: `s"Pushing data to $url: $json"`
- Otherwise log less info: `s"Pushing data to $url"`

Because `isTraceEnabled` is not exposed, currently I can't do the above.

Please expose `isTraceEnabled` etc.

**Comment by mattroberts297** [comment]
Please upgrade to `1.7.10`:

``` scala
libraryDependencies += "org.slf4s" %% "slf4s-api" % "1.7.10"
```

Then, given a `log`, you can do:

``` scala
if (log.underlying.isTraceEnabled) log.underlying.trace(s"Pushing data to $url: $json")
else log.info(s"Pushing data to $url")
```

You should use the underlying implementation otherwise you will have duplicate `isTraceEnabled` calls at runtime.

**Comment by mattroberts297** [comment]
Also, you could consider obtaining a separate Logger for your 'over-the-wire' output. Then, given a `HttpClient`, you can do:

``` scala
val log = LoggerFactory[HttpClient]
val wireLog = LoggerFactory(s"${classOf[HttpClient].getName}.wire")
```

This is what [Apache do](https://hc.apache.org/httpcomponents-client-ga/logging.html).

**Comment by ngocdaothanh** [comment]
Thanks!
So this `underlying` is not available in 1.7.7.
Can you add this `underlying` trick to README?

**Comment by mattroberts297** [comment]
Sadly, I didn't think to add it until 1.7.10 :disappointed:.

And of course! It should be there along with how to avoid duplicate calls.
